reproductions/SmokeTests.Core/SmokeTestNativeMethods.cs

```diff
@@ -0,0 +1,57 @@
+using System.Runtime.InteropServices;
+
+namespace SmokeTests.Core
+{
+    public class SmokeTestNativeMethods
+    {
+        public static bool IsProfilerAttachedExhaustive()
+        {
+            try
+            {
+                return RelativeDll.IsProfilerAttached();
+            }
+            catch
+            {
+                try
+                {
+                    return RelativeSo.IsProfilerAttached();
+                }
+                catch
+                {
+                    try
+                    {
+                        return DockerNativeProjectX64So.IsProfilerAttached();
+                    }
+                    catch
+                    {
+                        return DockerNativeProjectX86So.IsProfilerAttached();
```

Comment:
Can you try catch the final `IsProfilerAttached()` call and return false in the catch block?

Reply:
For smoke tests I'm totally okay with it bombing out if the profiler is not attached.
No strong opinion either way, I can do this.